docs/topics/rotate-certs.md

```diff
@@ -0,0 +1,127 @@
+
+1. You will need access to the API Model (`apimodel.json`) that was generated by `aks-engine deploy` (by default this file is placed into a relative directory that looks like `_output/<clustername>/`).
+
+1. A successful `aks-engine rotate-certs` operation can cause minimimal API Server downtime.
+
+1. `aks-engine rotate-certs` expects an API model that conforms to the current state of the cluster. `aks-engine rotate-certs` executes remote commands on the cluster nodes and uses the API Model information to establish a secure SSH connection. `aks-engine rotate-certs` also relies on some resources (such as VMs) to be named in accordance with the original `aks-engine` deployment.
+
+1. `aks-engine rotate-certs` relies upon a working connection to the cluster control plane during certificate rotation, both (1) to validate each step of the process, and (2) to restart/recreate cluster resources like `kube-system` pods and service account tokens. If you are rotating the certificates of a **private cluster**, you must run `aks-engine rotate-certs` from a host VM that has network access to the control plane, for example a jumpbox VM that resides in the same VNET as the master VMs. For more information on private clusters [refer to this documentation](features.md#feat-private-cluster).
+
+1. If using `aks-engine rotate-certs` in production, it is recommended to stage a certificate rotation test on an cluster that was built to the same specifications (built with the same cluster configuration + the same version of the `aks-engine` command line tool + the same set of enabled addons) as your production cluster before performing the certificate rotation, especially if the cluster configuration is "interesting", or in other words differs significantly from defaults. The reason for this is that AKS Engine supports many different cluster configurations and the extent of E2E testing that the AKS Engine team runs cannot practically cover every possible configuration. Therefore, it is recommended that you ensure in a staging environment that your specific cluster configuration works with `aks-engine rotate-certs` before attempting this potentially destructive operation on your production cluster.
+
+1. `aks-engine rotate-certs` does **not** guarantees backwards compatibility. If you deployed with `aks-engine` version `0.27.x`, you should prefer executing the certificate rotation process with version `0.27.x`.
+
+### Parameters
+
+|Parameter|Required|Description|
+|-----------------|---|---|
+|--api-model|yes|Relative path to the API model (cluster definition) that declares the expected cluster configuration.|
+|--location|yes|Azure location where the cluster is deployed.|
+|--ssh-host|yes|FQDN, or IP address, of an SSH listener that can reach all nodes in the cluster.|
+|--linux-ssh-private-key|yes|Path to a valid private SSH key to access the cluster's Linux nodes.|
+|--certificate-profile|no|Relative path to a JSON file containing the new set of certificates.|
```

Comment:
So this is not required because without it you're essentially telling the aks-engine binary to generate new certs itself based on info in the API model?

Reply:
partially yes, without it aks-engine binary will generate the new certs, if `--generate-new-certs` is pass.

I am now thinking that the behavior triggered by `--generate-new-certs` can be inferred.